### PR TITLE
CRYPTO-30: Mutable fields should be private

### PR DESCRIPTION
**Pull request by winningsix**
Mutable fields should be private or possibly package-protected, never protected or public.

### CODE REVIEW

File: src/main/java/org/apache/commons/crypto/stream/CipherInputStream.java [hunk 1]
```diff
@@ -430,6 +429,42 @@ protected Cipher getCipher() {
...
+  public byte[] getIv() {
+    return iv;
+  }
+
+  /**
+   * Set the IV.
+   *
+   * @param iv the IV.
+   */
+  public void setIv(byte[] iv) {
```
Comment (jerrychenhf) [inline src/main/java/org/apache/commons/crypto/stream/CipherInputStream.java]: Do we really need setIv method? it is passed in constructor and we may don't need to change it.

File: src/main/java/org/apache/commons/crypto/stream/CipherInputStream.java [hunk 2]
```diff
@@ -430,6 +429,42 @@ protected Cipher getCipher() {
...
+  public Input getInput() {
+    return input;
+  }
+
+  /**
+   * Set the input.
+   *
+   * @param input the input
+   */
+  public void setInput(Input input) {
```
Comment (jerrychenhf) [inline src/main/java/org/apache/commons/crypto/stream/CipherInputStream.java]: Do we need public setInput? the same as setIv.

File: src/main/java/org/apache/commons/crypto/stream/CipherOutputStream.java [hunk 3]
```diff
@@ -388,4 +386,112 @@ protected void freeBuffers() {
     Utils.freeDirectBuffer(inBuffer);
     Utils.freeDirectBuffer(outBuffer);
   }
+
```
Comment (jerrychenhf) [inline src/main/java/org/apache/commons/crypto/stream/CipherOutputStream.java]: We may need to consider whether we need a setXXX method to expose it to external change. If not allowed we should not expose the method.

**Comment by jerrychenhf** [comment]
Xianda has a bigger change to refine the parameters. So you keep work after Xianda's code merge or for time, we can work based on his code to avoid conflicts.

**Comment by jerrychenhf** [comment]
+1

**Comment by winningsix** [comment]
Thanks @jerrychenhf for your review. Committed to the upstream at 98bec4c

**Pull request closed without merging**